src/ImageSharp/Formats/Tga/TgaDecoderCore.cs

```diff
@@ -504,7 +591,7 @@ private void ReadRle<TPixel>(int width, int height, Buffer2D<TPixel> pixels, int
                 this.UncompressRle(width, height, bufferSpan, bytesPerPixel);
                 for (int y = 0; y < height; y++)
                 {
-                    int newY = Invert(y, height, inverted);
+                    int newY = InvertY(y, height, origin);
                     Span<TPixel> pixelRow = pixels.GetRowSpan(newY);
                     int rowStartIdx = y * width * bytesPerPixel;
                     for (int x = 0; x < width; x++)
```

Comment:
These per-pixel conditionals will be hurting performance also.

Reply:
i think i can move it outside the loop. Let me first get the tests green and then i will give it a try.

Reply:
i have moved the calculation of the x and y positions out of the loop where possible, see: 70a62ead9e6bf66e2b4b85d2212be832380d8c7b.

The only place where i could not manage to do so (without making it overly complicated), was the RLE parts.

Reply:
Yeah that's great 👍